build/test-aot.ps1

```diff
@@ -0,0 +1,50 @@
+param([string]$targetNetFramework)
+
+$projectName='Microsoft.Identity.Web.AotCompatibiliy.TestApp'
+$rootDirectory = Split-Path $PSScriptRoot -Parent
+$publishOutput = dotnet publish $rootDirectory/tests/$projectName/$projectName.csproj --framework $targetNetFramework -nodeReuse:false /p:UseSharedCompilation=false
+
+$actualWarningCount = 0
+
+foreach ($line in $($publishOutput -split "`r`n"))
+{
+    if (($line -like "*analysis warning IL*") -or ($line -like "*analysis error IL*"))
+    {
+        Write-Host $line
+        $actualWarningCount += 1
+    }
+}
+
+Write-Host "Actual warning count is: ", $actualWarningCount
+$expectedWarningCount = 41
+
+if ($LastExitCode -ne 0)
+{
+    Write-Host "There was an error while publishing AotCompatibility Test App. LastExitCode is:", $LastExitCode
+    Write-Host $publishOutput
+}
+
+$runtime = if ($IsWindows) {  "win-x64" } elseif ($IsMacOS) { "macos-x64"} else {"linux-x64"}
+$app = if ($IsWindows ) {"./$projectName.exe" } else {"./$projectName" }
+
+Push-Location $rootDirectory/tests/$projectName/bin/Release/$targetNetFramework/win-x64
+
+Write-Host "Executing test App..."
+$app
+Write-Host "Finished executing test App"
+
+if ($LastExitCode -ne 0)
+{
+  Write-Host "There was an error while executing AotCompatibility Test App. LastExitCode is:", $LastExitCode
+}
+
+Pop-Location
+
+$testPassed = 0
+if ($expectedWarningCount -lt $actualWarningCount)
```

Comment:
I would fail any time they aren't equal. So if someone is decreasing the warning count, they need to update the baseline.

Reply:
I was thinking of de-linking the fixing of warnings and update of this script. Keeping the link would make this script brittle. I will make this change.